### PR TITLE
Sys 4207 read contract from ethereum

### PR DESCRIPTION
**Pull request by micaelffrancoAV**
## Proposed changes

Introduces `read_bridge_contract` and `latest_finalised_ethereum_block` to the bridge interface.

## Type of change/Merge

🚨What type of change is this PR? </br>
_Put an `x` in the boxes that apply_

- [ ] Release 
  - [ ] Increase versions 
  - [ ] Baseline tests passed  
  - Release type:
    - [ ] Major release 
    - [ ] Minor release 
    - [ ] Patch release 

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._

- [ ] You describe the purpose of the PR, e.g.:
  - What does it do?
  - Highlight what important points reviewers should know about;
  - Indicates if there is something left for follow-up PRs.
- [ ] Documentation updated
- [ ] Business logic tested successfully
- [ ] Verify First, Write Last: In Substrate development, it is important that you always ensure preconditions are met and return errors at the beginning. After these checks have completed, then you may begin the function's computation.

### CODE REVIEW

File: pallets/eth-bridge/src/eth.rs [hunk 1]
```diff
@@ -258,6 +254,24 @@ fn make_ethereum_call<R, T: Config>(
     process_result(result)
 }
 
+pub fn make_historical_ethereum_call<R, T: Config>(
+    author_account_id: &T::AccountId,
+    endpoint: &str,
```
[the remark this answers is not in the document]
Comment (micaelffrancoAV) [reply pallets/eth-bridge/src/eth.rs:259]: Yes, we just need to add a new endpoint to the vow-service

**Review by nahuseyoum — APPROVED** [review]

File: primitives/avn-common/src/event_discovery.rs [hunk 2]
```diff
@@ -191,7 +191,7 @@ pub mod events_helpers {
     }
 
     // TODO unit test this
-    pub fn compute_finalised_block_number(
+    pub fn compute_adjusted_block_number(
```
Comment (thadouk) [inline primitives/avn-common/src/event_discovery.rs:194]:
```suggestion
    pub fn compute_start_block_from_finalised_block_number(
```

**Review by thadouk — COMMENTED** [review]
This PR is a bit confusing, it seems like its a preparation PR for a feature branch that does some renaming operation but also introduces some code that isn't used anywhere.
Since the target branch is main, I think we should either split the PR into two, or change the target to the feature branch that consumes these.

**Review by ivan-cholakov — APPROVED** [review]

File: pallets/avn/src/lib.rs [hunk 3]
```diff
@@ -766,6 +766,14 @@ pub trait BridgeInterface {
         params: &LowerParams,
         caller_id: Vec<u8>,
     ) -> Result<(), DispatchError>;
+    fn read_bridge_contract(
+        author_account_bytes: Vec<u8>,
+        function_name: &[u8],
+        params: &[(Vec<u8>, Vec<u8>)],
+        eth_block: Option<u32>,
```
[the remark this answers is not in the document]
Comment (micaelffrancoAV) [reply pallets/avn/src/lib.rs:773]: Added a Jira task for this: https://aventus-network-services.atlassian.net/browse/SYS-4276

**Review by nahuseyoum — APPROVED** [review]

**Pull request merged**